tests/unit/storage/data/DummyService.js

```diff
@@ -133,6 +134,42 @@ class DummyService {
         objStream.pipe(writeStream);
         return callback();
     }
+    putObjectTagging(tagParams, callback) {
+        if (tagParams.Key === 'externalBackendTestBucket/externalBackendMissingKey') {
+            const err = new Error();
+            err.code = 404;
+            err.message = 'NoSuchKey';
+            return callback(err);
+        }
+
+        const keys = Object.keys(tagParams);
+        assert.ok(keys.length > 0);
+        assert.ok(tagParams.Tagging.TagSet.length > 0);
+        tagParams.Tagging.TagSet.forEach(tag => {
+            assert.ok(tag.Key.length > 0);
+            assert.ok(tag.Value.length > 0);
+        });
+
+        if (tagParams.VersionId) {
+            assert.ok(tagParams.VersionId === 'latestversion');
+        }
+
+        return callback();
+    }
+    deleteObjectTagging(tagParams, callback) {
+        if (tagParams.Key === 'externalBackendTestBucket/externalBackendMissingKey') {
+            const err = new Error();
```

Comment:
You should be able to use errors.NoSuchKey for this, look at errors/arsenalErrors.json